File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.c


int minCostClimbingStairs(int* cost, int costSize){
    int a = 0,b = 0;
    for(int i=2; i <= costSize; ++i) {
        int v1 = a + cost[i-2];
        int v2 = b + cost[i-1];
        a = b;
        b = v1 < v2 ? v1 : v2;
    }
    return b;

}

File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        n = len(cost)
        
        dp = {}
        dp[0] = 0
        dp[1] = 0
        
        for i in range(2,n+1):
            v1 = dp[i-1] + cost[i-1]
            v2 = dp[i-2] + cost[i-2]
            dp[i] = v1 if v1 < v2 else v2
        
        return dp[n]